### PR TITLE
Types

### PR DESCRIPTION
**Pull request by oddcoder**

### CODE REVIEW

**Comment by radare** [comment]
if you indent a file use the sys/indent-whitelist.sh (and add your file in there), so we can slowly verify that everything is properly indented

**Comment by oddcoder** [comment]
well it showed that my file is indented well :)

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -61,7 +61,7 @@ static int sdbdeletelink(void *p, const char *k, const char *v) {
 }
 static int linklist(void *p, const char *k, const char *v){
 	if(!strncmp(k,"link.",strlen("link.")))
-		r_cons_printf("tl %s @ %s\n",v,k+strlen("link."));
+		r_cons_printf("tl %s @ 0x%s\n",v,k+strlen("link."));
```
Comment (radare) [inline libr/core/cmd_type.c]: wrong indent. missing spaces

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -61,7 +61,7 @@ static int sdbdeletelink(void *p, const char *k, const char *v) {
 }
 static int linklist(void *p, const char *k, const char *v){
 	if(!strncmp(k,"link.",strlen("link.")))
```
Comment (radare) [inline libr/core/cmd_type.c]: more missing spaces

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -312,10 +312,11 @@ static int cmd_type(void *data, const char *input) {
...
 			const char *help_message[] = {
 				"Usage:", "", "",
-				" tl", " <typename>", "link a type to current adress.",
-				" tl", " <typename> <address>", "link type to given address.",
-				" tl-*", "", " delete all links.",
+				" tl", "<typename>", "link a type to current adress.",
+				" tl", "<typename> <address>", "link type to given address.",
+				" tl-*", "", "delete all links.",
 				" tl-", "<address>", "delete link at given address.",
+				"tl*","","list all links in radare2 command format",
```
Comment (radare) [inline libr/core/cmd_type.c]: why that space on every single command?

File: libr/core/cmd_type.c [hunk 4]
```diff
@@ -330,7 +330,7 @@ static int cmd_type(void *data, const char *input) {
 				addr = r_num_math (core->num, ptr);
 			} else addr = core->offset;
 			r_anal_type_link (core->anal, type, addr);
-			} break;
```
Comment (radare) [inline libr/core/cmd_type.c]: this is wrong indentation. bear in mind that indent.sh is not 100% correct

File: libr/core/cmd_type.c [hunk 5]
```diff
@@ -340,11 +340,11 @@ static int cmd_type(void *data, const char *input) {
 				const char *ptr = input + 3;
 				ut64 addr = r_num_math (core->num, ptr);
 				r_anal_type_unlink (core->anal, addr);
-				} break;
+			} break;
 			}
 			break;
 		case '*':
-			sdb_foreach (core->anal->sdb_types,linklist,core);
+			sdb_foreach (core->anal->sdb_types, linklist, core);
```
Comment (radare) [inline libr/core/cmd_type.c]: those indent fixes are good, but you should squash commits if one of them just "indents"

File: libr/core/cmd_type.c [hunk 6]
```diff
@@ -348,6 +349,17 @@ static int cmd_type(void *data, const char *input) {
...
+		const char *type = input + 2;
+		char *ptr = strchr (type, ' ');
+		*ptr++ = 0;
+		ut64 addr = r_num_math (core->num, ptr);
+		char *fmt = r_anal_type_format (core->anal, type);
+		if (fmt) {
+			r_core_cmdf (core, "pf %s @ 0x%08" PFMT64x "\n", fmt, addr);
+			free (fmt);
+		} else eprintf ("Cannot find '%s' type\n", input + 1);
+	} break;
```
Comment (radare) [inline libr/core/cmd_type.c]: wrong indent. that bracket must be at the same level as the opened one. one tab right

Comment (radare) [inline libr/core/cmd_type.c]: the break goes for the case, not the block, it may probably confuse the indentator

**Comment by radare** [comment]
Merged and fixed!

**Pull request closed without merging**